src/inet/common/misc/ThruputMeter.cc

```diff
@@ -46,7 +46,9 @@ void ThruputMeter::initialize()
 
 void ThruputMeter::handleMessage(cMessage *msg)
 {
-    updateStats(simTime(), PK(msg)->getBitLength());
+    auto pkt = dynamic_cast<cPacket *>(msg);
```

Comment:
Should use msg->isPacket() to check for being a packet, it's faster than dynamic_cast.

Reply:
Ok, this simplify the if